spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/core/SpannerTemplateTests.java

```diff
@@ -287,49 +284,48 @@ public void readOnlyTransactionTest() {
     verify(readOnlyTransaction, times(2)).read(eq("custom_test_table"), any(), any());
   }
 
-  @Test
-  public void readOnlyTransactionDmlTest() {
 
-    this.expectedException.expectMessage("A read-only transaction template cannot execute DML.");
+  @Test
+  <T> void readOnlyTransactionDmlTest() {
 
     ReadOnlyTransaction readOnlyTransaction = mock(ReadOnlyTransaction.class);
     when(this.databaseClient.readOnlyTransaction(
             // exact staleness is expected.
             TimestampBound.ofReadTimestamp(Timestamp.ofTimeMicroseconds(333))))
         .thenReturn(readOnlyTransaction);
 
-    this.spannerTemplate.performReadOnlyTransaction(
-        spannerOperations -> {
-          spannerOperations.executeDmlStatement(Statement.of("fail"));
-          return null;
-        },
-        new SpannerReadOptions().setTimestamp(Timestamp.ofTimeMicroseconds(333)));
+    SpannerReadOptions testSpannerReadOptions = new SpannerReadOptions().setTimestamp(Timestamp.ofTimeMicroseconds(333));
+    Function<SpannerTemplate, T> testSpannerOperations = spannerOperations -> {
+      spannerOperations.executeDmlStatement(Statement.of("fail"));
+      return null;
+    };
+
+    assertThatThrownBy(() -> this.spannerTemplate.performReadOnlyTransaction(
+            testSpannerOperations, testSpannerReadOptions))
+            .hasMessage("A read-only transaction template cannot execute DML.");
   }
 
   @Test
-  public void readOnlyTransactionPartitionedDmlTest() {
-
-    this.expectedException.expectMessage(
-        "A read-only transaction template cannot execute partitioned" + " DML.");
+  <T> void readOnlyTransactionPartitionedDmlTest() {
 
     ReadOnlyTransaction readOnlyTransaction = mock(ReadOnlyTransaction.class);
     when(this.databaseClient.readOnlyTransaction(
             TimestampBound.ofReadTimestamp(Timestamp.ofTimeMicroseconds(333))))
         .thenReturn(readOnlyTransaction);
 
-    this.spannerTemplate.performReadOnlyTransaction(
-        spannerOperations -> {
-          spannerOperations.executePartitionedDmlStatement(Statement.of("fail"));
-          return null;
-        },
-        new SpannerReadOptions().setTimestamp(Timestamp.ofTimeMicroseconds(333)));
+    SpannerReadOptions testSpannerReadOptions = new SpannerReadOptions().setTimestamp(Timestamp.ofTimeMicroseconds(333));
+    Function<SpannerTemplate, T> testSpannerOperations = spannerOperations -> {
+      spannerOperations.executePartitionedDmlStatement(Statement.of("fail"));
+      return null;
+    };
+
+    assertThatThrownBy(() -> this.spannerTemplate.performReadOnlyTransaction(
+            testSpannerOperations, testSpannerReadOptions))
+            .hasMessage("A read-only transaction template cannot execute partitioned" + " DML.");
```

Comment:
Could you remove the concatenation while you are here? I looked into how this came to be, and it was a result of original formatting being on two lines, and then getting joined to a single line when I ran autoformatting to google style.